com.reprezen.swagedit/src/com/reprezen/swagedit/validation/SwaggerError.java

```diff
@@ -109,20 +128,66 @@ public String getMessage() {
 			builder.append("Failed to match exactly one schema:");
 			builder.append("\n");
 
-			for (String key: errors.keySet()) {
+			for (String location : orderedErrorLocations) {
 				builder.append(tabs);
 				builder.append(" - ");
-				builder.append(key);
+				builder.append(getHumanFriendlyText(location));
 				builder.append(":");
 				builder.append("\n");
 
-				for (SwaggerError e: errors.get(key)) {
+				for (SwaggerError e : errors.get(location)) {
 					builder.append(e.getMessage(true));
 				}
 			}
 
 			return builder.toString();
 		}
+
+		protected String getHumanFriendlyText(String location) {
+			JsonNode swaggerSchemaNode = findNode(location);
+			if (swaggerSchemaNode == null) {
+				return location;
+			}
+			JsonNode title = swaggerSchemaNode.get("title");
+			if (title != null) {
+				return title.asText();
+			}
+			JsonNode description = swaggerSchemaNode.get("title");
```

Comment:
Was this supposed to be `.get("description")` ?  I'm not sure using the description is a good idea.  We might be better off putting titles in wherever they might be omitted from the original swagger schema.